monky.el

```diff
@@ -1327,22 +1441,19 @@ With a prefix argument, visit in other window."
 
 ;;; Updating
 
-(defun monky-fetch ()
-  "Run hg fetch."
-  (interactive)
-  (let ((remote (if current-prefix-arg
-                    (monky-read-remote "Fetch from : ")
-                  monky-incoming-repository)))
-    (monky-run-hg-async "fetch" remote
-                        "--config" "extensions.fetch=")))
-
 (defun monky-pull ()
-  "Run hg pull."
+  "Run hg pull.  If monky-run-update-after-pull is t, also run hg update."
   (interactive)
   (let ((remote (if current-prefix-arg
                     (monky-read-remote "Pull from : ")
                   monky-incoming-repository)))
-    (monky-run-hg-async "pull" remote)))
+    (cond (monky-run-update-after-pull
+	   (monky-run-hg-sync "pull" remote)
+	   (monky-run-hg-async "update" "--tool" "internal:merge"))
+	  (monky-run-rebase-after-pull
+	   (monky-run-hg-sync "pull" remote)
+	   (monky-run-hg-async "rebase" "--tool" "internal:merge"))
```

Comment:
```rebase``` must be run in single command with ```pull```

In case of separate commands repo remains unupdated:
```
❯ hg pull
pulling from ...
added 1 changesets with 1 changes to 1 files
new changesets 4e2a367469f7
...
(run 'hg update' to get a working copy)

❯ hg rebase
abort: branch '...' has one head - please rebase to an explicit rev
(run 'hg heads' to see all heads)
```

In case of single command repo is updated if there are nothing to rebase:
```
❯ hg pull --rebase
pulling from ...
added 1 changesets with 1 changes to 1 files
new changesets a92cb7c91ea5
nothing to rebase - updating instead
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
```

Also i don't like hardcoded merge tool in these commands. I have ```merge=:merge3``` in my ```.hgrc``` and i can't see any reason to override this setting on ```pull```. 

Let's run ```pull``` like this:
```
(monky-run-hg-sync "pull" "--update" remote)
```

And for rebase:
```
(monky-run-hg-sync "pull" "--rebase" remote)
```

Reply:
Maybe it would be better to define customizable variable like ```monky-pull-args``` and one can set it to ```--update``` or ```--rebase``` instead of two variables ```monky-run-rebase-after-pull``` and ```monky-run-update-after-pull```